### PR TITLE
create `whiteout` when renaming file over existing `whiteout`

### PR DESCRIPTION
**Pull request by flouthoc**
Create `whiteout` for cases if there is already a whiteout in the destination path
and rename is called with flag `RENAME_NOREPLACE`.

Close: https://github.com/containers/fuse-overlayfs/issues/354

### CODE REVIEW

**Comment by flouthoc** [comment]
This fixes the reported issue but i have to discuss the patch first and it would be nice if @Tuupertunut could try this as well.

File: main.c [hunk 1]
```diff
@@ -4631,9 +4631,12 @@ ovl_rename_direct (fuse_req_t req, fuse_ino_t parent, const char *name,
       ret = direct_renameat2 (srcfd, name, destfd,
                               newname, flags|RENAME_WHITEOUT);
     }
-      /* If the destination is a whiteout, just overwrite it.  */
-  if (ret < 0 && errno == EEXIST)
-    ret = direct_renameat2 (srcfd, name, destfd, newname, flags & ~RENAME_NOREPLACE);
+  /* If the destination is a whiteout, just overwrite it.  */
+  if (ret < 0 && errno == EEXIST) {
+    ret = direct_renameat2(srcfd, name, destfd, newname, flags & ~RENAME_NOREPLACE);
+    if (!(ret < 0))
```
Comment (giuseppe) [inline main.c:4637]: this looks weird.

Could we have `ret >= 0`?

File: main.c [hunk 2]
```diff
@@ -4631,9 +4631,12 @@ ovl_rename_direct (fuse_req_t req, fuse_ino_t parent, const char *name,
...
+      create_whiteout(lo, pnode, name, false, true);
```
Comment (giuseppe) [inline main.c:4638]: don't we need to check the return code?

And why do we need `force_create`?

File: main.c (hunk 1 above)
Comment (flouthoc) [reply main.c]: fixed.

**Comment by Tuupertunut** [comment]
From the code it looks like this fixes the issue. However I have already done a larger change for the rename code that also encompasses this fix and I will make a PR for it, you may want to check that instead.

**Comment by flouthoc** [comment]
> From the code it looks like this fixes the issue. However I have already done a larger change for the rename code that also encompasses this fix and I will make a PR for it, you may want to check that instead.

@Tuupertunut Sure we can close this one if that PR already addresses this.

**Comment by Tuupertunut** [comment]
Done: #357

**Comment by flouthoc** [comment]
Closing this one in favor of : https://github.com/containers/fuse-overlayfs/pull/357

**Pull request closed without merging**